### PR TITLE
refac: create a workspace

### PR DESCRIPTION
**Pull request by sayon**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

File: era-compiler-solidity/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Contribution Guidelines
+
+Hello! Thanks for your interest in joining the mission to accelerate the mass adoption of crypto for personal
```
Comment (hedgar2017) [inline era-compiler-solidity/CONTRIBUTING.md:3]: All this must stay at the root.
Only `src` and `Cargo.toml` go to the crate directory.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: era-compiler-solidity/.cargo/config.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[tools.clippy]
```
Comment (hedgar2017) [inline era-compiler-solidity/.cargo/config.toml:1]: Does it only work this way?
Maybe the root file is enough.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: era-compiler-solidity/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[package]
+name = "era-compiler-solidity"
+version = "1.5.2"
+authors = [
+    "Oleksandr Zarudnyi <a.zarudnyy@matterlabs.dev>",
+]
+license = "MIT OR Apache-2.0"
```
Comment (antonbaliasnikov) [inline era-compiler-solidity/Cargo.toml:7]: I believe `license` and `edition` fields can go to the workspace level to not be duplicated in packages? Or we expect different ones?

Comment (hedgar2017) [reply era-compiler-solidity/Cargo.toml:7]: Usually we keep those on crate level to avoid copy-pasting later.
@sayon you may add yourself here :)

Comment (antonbaliasnikov) [reply era-compiler-solidity/Cargo.toml:7]: Here is an example:
https://github.com/matter-labs/era-compiler-common/blob/main/solc-downloader/Cargo.toml#L4-L6
https://github.com/matter-labs/era-compiler-common/blob/main/Cargo.toml

File: era-compiler-solidity/.cargo/config.toml (hunk 2 above)
Comment (sayon) [reply era-compiler-solidity/.cargo/config.toml:1]: It is, i just checked. Will move it to the root one

**Review by hedgar2017 — APPROVED** [review]

**Pull request merged**